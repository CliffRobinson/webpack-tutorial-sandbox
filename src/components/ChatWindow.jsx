import React, { useEffect } from 'react'
import log from 'loglevel'


export function ChatWindow(props) {
    const { chat, requestMessagesByRoom, updateCurrentMessage, addMessage, room_id } = props
    const user_id = 1
    const {messages, currentMessage} = chat
    const styleObj = {
        backgroundColor: 'white',
        border: "2px solid black"
    }

    function clicker(e) {
        e.preventDefault()
        const message = {
            room_id,
            user_id,
            time: new Date().getTime(),
            msg: currentMessage
        }
        addMessage(message)
    }

    function onInputChange(e) {
        updateCurrentMessage(e.target.value)
    }

    useEffect(() => {
        log.trace('chatwindow component is dispatching requestmessages action')
        requestMessagesByRoom(room_id)
    },[])

    return (
        <form>

                <div style={styleObj}>
                    {messages.map((msg, i) => <p key={i}>{msg.name}: {msg.msg}</p>)}
                </div>
                <button onClick={clicker}>Get memssages</button>
                <input type='text'onChange={onInputChange}></input>


        </form>
    )

}