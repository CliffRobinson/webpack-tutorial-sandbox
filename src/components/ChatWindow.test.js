import React from 'react'
import { shallow } from 'enzyme'

import { ChatWindow } from './ChatWindow'

const data = {
    currentMessage: "",
    messages: [
        {
            name: 'Harry',
            msg: 'db: SANPE KILLS DUMBELDORE!!',
            time: 1585802175899
        },
        {
            name: 'Ron',
            msg: 'db: idk why I hang out with you',
            time: 1585802175950
        }
    ]
}

test('chatWindow', () => {
    expect(shallow(<ChatWindow chat={data} />)).toMatchSnapshot()
})