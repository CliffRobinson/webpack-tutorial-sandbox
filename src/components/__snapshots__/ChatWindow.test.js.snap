// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chatWindow 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "border": "2px solid black",
    }
  }
>
  <p
    key="0"
  >
    : 
    db: SANPE KILLS DUMBELDORE!!
  </p>
  <p
    key="1"
  >
    : 
    db: idk why I hang out with you
  </p>
</div>
`;
